src/lib/agent/LedgerService.ts

```diff
@@ -0,0 +1,181 @@
+import logger from '../logger';
+import { Wallet } from '../wallet/Wallet';
+
+export class LedgerService {
+  wallet: Wallet;
+  indy: Indy;
+  poolHandle?: PoolHandle;
+  authorAgreement?: AuthorAgreement;
+
+  constructor(wallet: Wallet, indy: Indy) {
+    this.wallet = wallet;
+    this.indy = indy;
+  }
+
+  async connect(poolName: string, poolConfig: PoolConfig) {
+    try {
+      logger.log(__dirname);
+
+      logger.log('Creating pool config');
+      await this.indy.createPoolLedgerConfig(poolName, poolConfig);
+    } catch (e) {
+      logger.log('PoolLedgerConfigAlreadyExists');
+      if (e.message !== 'PoolLedgerConfigAlreadyExistsError') {
+        throw e;
+      }
+    }
+
+    logger.log('Setting protocol version');
+    await this.indy.setProtocolVersion(2);
+
+    logger.log('Opening pool');
+    this.poolHandle = await this.indy.openPoolLedger(poolName);
+  }
+
+  async getPublicDid(myDid: Did) {
+    if (!this.poolHandle) {
+      throw new Error('Pool has not been initialized.');
+    }
+    const request = await this.indy.buildGetNymRequest(null, myDid);
+    logger.log('request', request);
+
+    const response = await this.indy.submitRequest(this.poolHandle, request);
+    logger.log('response', response);
+
+    const result = await this.indy.parseGetNymResponse(response);
+    logger.log('result', result);
+
+    return result;
+  }
+
+  async registerSchema(myDid: Did, schemaTemplate: SchemaTemplate): Promise<[SchemaId, Schema]> {
+    if (!this.poolHandle) {
+      throw new Error('Pool has not been initialized.');
+    }
+    const { name, attributes, version } = schemaTemplate;
+    const [schemaId, schema] = await this.indy.issuerCreateSchema(myDid, name, version, attributes);
+    logger.log('schemaId, schema', schemaId, schema);
+
+    const request = await this.indy.buildSchemaRequest(myDid, schema);
+    logger.log('request', request);
+
+    const requestWithTaa = await this.appendTaa(myDid, request);
+
+    const signedRequest = await this.wallet.signRequest(myDid, requestWithTaa);
+    logger.log('signedRequest', signedRequest);
```

Comment:
Here we could add more detail to the log messages.

Just an example, would not use this directly:

```suggestion
    logger.log(`registering schema with schemaId: ${schemaId}`, schema);

    const request = await this.indy.buildSchemaRequest(myDid, schema);
    logger.log('created create schema request', request);

    const requestWithTaa = await this.appendTaa(myDid, request);

    const signedRequest = await this.wallet.signRequest(myDid, requestWithTaa);
    logger.log('signed create schema request', signedRequest);
```

Reply:
Updated